components/mdns/mdns.c

```diff
@@ -569,7 +569,9 @@ static const uint8_t * _mdns_read_fqdn(const uint8_t * packet, const uint8_t * s
                     && (strcmp(buf, MDNS_DEFAULT_DOMAIN) != 0)
                     && (strcmp(buf, "ip6") != 0)
                     && (strcmp(buf, "in-addr") != 0)) {
-                snprintf((char*)name, MDNS_NAME_BUF_LEN, "%s.%s", name->host, buf);
+              //snprintf((char*)name, MDNS_NAME_BUF_LEN, "%s.%s", name->host, buf);
+                strlcat(name->host, (char*)'.', sizeof(name->host));
```

Comment:
I think you mean `"."` instead of `(char *)'.'`. The latter is a pointer to the ASCII value of dot, which is certainly not valid memory. :)

Reply:
Certainly, fixed ;)